### ISSUE TITLE
Assert a function asserts?

### ISSUE THREAD

**Issue by tomc1998** [bitwes/Gut#100]
Hey, is there any way to test that a function asserts false? i.e.

```
func div(x, y):
    assert(y != 0)
    return x / y

func test_div():
    assert_eq(div(8, 4), 2)
    assert_asserts(div(1, 0))
```

If not currently, is this possible with gdscript currently?

**Comment by bitwes** [comment]
I thought maybe you could with the new functionality I'm about to release but `assert` is a reserved word.  It isn't part of `Object` or any other Built-In class so there is no way to hook into it.  It is part of the actual GDScript language the same way that `if` and `for` are.  I looked around, hoping that maybe I could get a count of `assert` calls that have passed/failed but there doesn't appear to be a way to do this.

Basically anything in GDScript cannot be stubbed/spied/doubled. 
 https://docs.godotengine.org/en/3.0/classes/class_@gdscript.html

**Comment by tomc1998** [comment]
How easy is it to plug into gdscript with the c++ plugins? Or is there really no way this is happening;)

**Comment by AlexAndDraw** [comment]
> How easy is it to plug into gdscript with the c++ plugins? Or is there really no way this is happening;)

Not easy at all to be frank. Asserts are debug only in godot anyway (they don't get compiled in the release version). My best advice would be to change them to an if and test for that. You could probably write your own assert static method if the crashing is necessary and have your export templates (or a code reformatter in VSCode etc) to clear them out before compiling your release version.

**Comment by tomc1998** [comment]
Hmm that's a shame. Thanks anyway!
